asreview/webapp/authentication/decorators.py

```diff
@@ -81,3 +90,38 @@ def decorated_function(*args, **kwargs):
         return f(projects, *args, **kwargs)
 
     return decorated_function
+
+
+def login_remote_user(f):
+    @wraps(f)
+    def decorated_function(*args, **kwargs):
+        if not current_user.is_authenticated:
+            remote_user_handler = current_app.config.get("REMOTE_USER", False)
+
+            if isinstance(remote_user_handler, RemoteUserHandler):
+                user_info = remote_user_handler.handle_request(request.environ)
+
+                if user_info["identifier"]:
+                    user = User.query.filter(
+                        User.identifier == user_info["identifier"]
+                    ).one_or_none()
+                    if not user:
+                        try:
+                            user = User(
+                                **user_info,
+                                origin="remote",
+                                public=True,
+                                confirmed=True,
+                            )
+                            DB.session.add(user)
+                            DB.session.commit()
+                        except (IntegrityError, SQLAlchemyError) as e:
+                            DB.session.rollback()
+                            error_500(e)
+
+                    login_user(
+                        user, remember=True, duration=datetime.timedelta(days=31)
```

Comment:
Should the duration be configurable?

Reply:
:+1: I've had to refactor the `perform_login_user` function for this.